File: projects/x-forth.c/src/core/define.c
#include "index.h"

definition_t *
define_constant(mod_t *mod, const char *name, value_t *value) {
    definition_t *definition =
        make_constant_definition(mod, string_copy(name), value);
    mod_define(mod, name, definition);
    return definition;
}

definition_t *
define_variable(mod_t *mod, const char *name, value_t *value) {
    definition_t *definition =
        make_variable_definition(mod, string_copy(name), value);
    mod_define(mod, name, definition);
    return definition;
}

File: projects/x-lisp-compiler.js/src/lisp/define/define.ts
import * as Definitions from "../definition/index.ts"
import { type Mod } from "../mod/index.ts"

export function definePrimitive(
  mod: Mod,
  name: string,
  options: { arity: number },
): void {
  mod.definitions.set(
    name,
    Definitions.PrimitiveDefinition(mod, name, options.arity),
  )
}
